File: app/models/article.py
"""Article model."""

from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import relationship

from app.core.datetime import utc_now
from app.db.database import Base


class Article(Base):
    """Article table with author ownership and category binding."""

    __tablename__ = "articles"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(200), index=True, nullable=False)
    summary = Column(String(500), nullable=True)
    content = Column(Text, nullable=False)
    view_count = Column(Integer, default=0, nullable=False)
    author_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    category_id = Column(
        Integer,
        ForeignKey("categories.id"),
        nullable=False,
        index=True,
    )
    created_at = Column(DateTime, default=utc_now, nullable=False)
    updated_at = Column(
        DateTime,
        default=utc_now,
        onupdate=utc_now,
        nullable=False,
    )

    author = relationship("User", back_populates="articles")
    category = relationship("Category", back_populates="articles")

File: frontend/src/types/article.ts
import type { CategorySimple } from "./category";

export type ArticleAuthor = {
  id: number;
  username: string;
};

export type ArticleListItem = {
  id: number;
  title: string;
  summary: string | null;
  author: ArticleAuthor;
  category: CategorySimple;
  created_at: string;
};

export type ArticleDetail = ArticleListItem & {
  content: string;
  view_count: number;
  updated_at: string;
};

export type ArticleCreateRequest = {
  title: string;
  summary?: string | null;
  content: string;
  category_id: number;
};

export type ArticleUpdateRequest = Partial<ArticleCreateRequest>;
